planet/cli/data.py

```diff
@@ -399,8 +402,109 @@ async def search_update(ctx,
+              help=('Overwrite files if they already exist.'))
+@click.option('--checksum',
+              is_flag=True,
+              default=None,
+              help=('Verify that checksums match.'))
+async def asset_download(ctx, asset, directory, filename, overwrite, checksum):
+    """Download an activated asset.
+
+    This function will fail if the asset state is not activated. Consider
+    calling `asset-wait` before this command to ensure the asset is activated.
+
+    If --checksum is provided, the associated checksums given in the manifest
+    are compared against the downloaded files to verify that they match.
+
+    If --checksum is provided, files are already downloaded, and --overwrite is
+    not specified, this will simply validate the checksums of the files against
+    the manifest.
+    """
+    quiet = ctx.obj['QUIET']
+    async with data_client(ctx) as cl:
+        path = await cl.download_asset(asset=asset,
+                                       filename=filename,
+                                       directory=Path(directory),
+                                       overwrite=overwrite,
+                                       progress_bar=not quiet)
+        if checksum:
+            cl.validate_checksum(asset, path)
+
+
+@data.command()
+@click.pass_context
+@translate_exceptions
+@coro
+@click.argument("asset", type=types.JSON())
+async def asset_activate(ctx, asset):
+    '''Activate an asset.'''
+    async with data_client(ctx) as cl:
+        await cl.activate_asset(asset)
+
+
+@data.command()
+@click.pass_context
+@translate_exceptions
+@coro
+@click.argument("asset", type=types.JSON())
+@click.option('--delay',
+              type=int,
+              default=5,
+              help='Time (in seconds) between polls.')
+@click.option('--max-attempts',
+              type=int,
+              default=200,
+              show_default=True,
+              help='Maximum number of polls. Set to zero for no limit.')
+async def asset_wait(ctx, asset, delay, max_attempts):
+    '''Wait for an asset to be activated.
+
+    Returns when the asset state has reached "activated" and the asset is
+    available.
+    '''
+    quiet = ctx.obj['QUIET']
+    async with data_client(ctx) as cl:
+        with StateBar(order_id="my asset", disable=quiet) as bar:
+            state = await cl.wait_asset(asset,
+                                        delay,
+                                        max_attempts,
+                                        callback=bar.update_state)
+    click.echo(state)
+
+
+@data.command()
+@click.pass_context
+@translate_exceptions
+@coro
+@click.argument("item_types")
+@click.argument("item_id")
+@click.argument("asset_type_id")
+@pretty
+async def asset_get(ctx, item_types, item_id, asset_type_id, pretty):
+    '''Get an item asset.'''
+    async with data_client(ctx) as cl:
+        asset = await cl.get_asset(item_types, item_id, asset_type_id)
+        echo_json(asset, pretty)
```

Comment:
ok this is a nit since it absolutely works either way, but echo_json doesn't need to be within the context of data_client and we have been calling it outside the context (at least when I look at the orders cli)
```suggestion
    echo_json(asset, pretty)
```